### DISCUSSION TITLE
Using Array.Empty instead of zero length byte array allocation

### DISCUSSION THREAD

**Discussion by fisherman6v6** [jchristn/WatsonTcp#227] [Ideas]
Hi!

Wouldn't be better to use `Array.Empty` instead of `new byte[0]` in send functions to avoid unnecessary allocations?

**Comment by jchristn** [comment]
Wouldn't a conversion to ```byte[]``` then be required?

**Reply by fisherman6v6** [reply]
Nope, since we can use the generic version` Array.Empty<byte>` as stated in this CA [https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1825](url).

**Reply by jchristn** [reply]
Thank you so much for sharing this with me!  I was completely unaware of this.  I just published v5.0.11 to NuGet, which addresses this.  I really appreciate you making the library better!  

NuGet: https://www.nuget.org/packages/WatsonTcp/5.0.11
Commit: https://github.com/jchristn/WatsonTcp/commit/a4b108512937aaa585ba97f067b29a1bac617b84
